Automattic/wp-calypso
# Plans 2023: Domain upsell plans page redone

Fixes Automattic/martech#1551
Related to https://github.com/Automattic/martech/issues/1491
Based off https://github.com/Automattic/wp-calypso/pull/73421 & https://github.com/Automattic/wp-calypso/pull/73418
## Proposed Changes

Various updates for the domain upsell (`/plans` page with a `?get_domain` parameter) redesign. 
- Issue https://github.com/Automattic/martech/issues/1491
- Figma xkhhUlDowF13dRoXJqlRDM-fi-1:95 

cc @lucasmendes-design 

## Media

### old plans grid

<img width="700" alt="Screenshot 2023-02-16 at 1 17 16 PM" src="https://user-images.githubusercontent.com/1705499/219367163-330c8bf2-d1fe-4d12-bf38-ac0d0d31323a.png">

### new plans grid

<img width="700" alt="Screenshot 2023-02-16 at 1 18 17 PM" src="https://user-images.githubusercontent.com/1705499/219367224-5c6b745e-a5eb-40ed-bcaf-775df68e84af.png">

### clicking on "skip" or "keep my plan"

<img width="700" alt="Screenshot 2023-02-16 at 1 31 01 PM" src="https://user-images.githubusercontent.com/1705499/219367278-ba2d45e3-1712-4bd7-8a77-b843e6e96b61.png">

## Testing Instructions



* Go to `plans/[SITE]?get_domain=foobar` with and without the feature flag (`&flags=-onboarding/2023-pricing-grid`). This is equivalent to clicking on the Home page pannel.
* Confirm the behavior outlined in the issue (screenshots above).
* Confirm that everything else is sane without the `get_domain` parameter.

## Pre-merge Checklist
